### PR TITLE
Make only live website indexable

### LINKED ISSUE #27338: Search engines indexing old releases
**Problem:**

Google search for "kubernetes secrets v2" returns:

image

(notice the URL subdomains)

**Proposed Solution:**

I need to do some more research but a few things come to mind:
- `enableRobotsTXT = true` is in [config.toml](https://github.com/kubernetes/website/blob/master/config.toml#L11) and set in the couple `release-1.xx` repos I checked.
- `X-Robots-Tag: noindex` is set in [layouts/index.headers](https://github.com/kubernetes/website/blob/master/layouts/index.headers#L37). I believe this is for non-prod sites that are running, however I believe even the old released sites are considered "production"

That is assuming robots tag is the culprit. What's also strange is that the above robot config is set in `release-1.19` but doesn't appear in the search results

I also think looking at [old robot TXT issues](https://github.com/kubernetes/website/search?q=robots.txt&type=issues) might help. It's possible this problem has been solved and phased out with future releases.

/priority important-soon
/triage accepted
/help
/cc @reylejano

### PR DESCRIPTION
**Pull request by sftim**
This change updates how we run Hugo **and** changes the logic for checking whether a page should be indexable (copied with a backport from upstream Docsy).

Let's not apply this change incautiously. It's important, *and* it's important to get right.
So I'll insta-hold it to highlight that:
/hold

Fixes #27338 (I hope)

### CODE REVIEW

**Comment by kbhawkey** [comment]
:eyes:

File: Makefile [hunk 1]
```diff
@@ -28,28 +28,30 @@ module-init:
...
 
 functions-build:
 	$(NETLIFY_FUNC) build functions-src
 
 check-headers-file:
 	scripts/check-headers-file.sh
 
-production-build: build check-headers-file ## Build the production site and ensure that noindex headers aren't added
+production-build: ## Build the production site and ensure that noindex headers aren't added
+	hugo --minify --environment production
```
Comment (kbhawkey) [inline Makefile:47]: Should this target call `module-check`?

File: Makefile [hunk 2]
```diff
@@ -70,10 +72,10 @@ container-image: ## Build a container image for the preview of the website
 		--build-arg HUGO_VERSION=$(HUGO_VERSION)
 
 container-build: module-check
-	$(CONTAINER_RUN) --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 $(CONTAINER_IMAGE) sh -c "npm ci && hugo --minify"
+	$(CONTAINER_RUN) --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 $(CONTAINER_IMAGE) sh -c "npm ci && hugo --minify ----buildFuture --environment development"
```
Comment (kbhawkey) [inline Makefile:75]: Check the `buildFuture` flag?

File: layouts/_default/baseof.html [hunk 3]
```diff
@@ -1,6 +1,9 @@
 <!doctype html>
 <html lang="{{ .Site.Language.Lang }}" class="{{.Params.class}} no-js">
-  <head>
+{{- if eq hugo.Environment "preview" -}}
+  <!-- deploy preview -->
```
Comment (kbhawkey) [inline layouts/_default/baseof.html:4]: Is this clause needed?

Comment (sftim) [reply layouts/_default/baseof.html:4]: It helps us check we're seeing a deploy preview build.

File: Makefile (hunk 2 above)
Comment (sftim) [reply Makefile:75]: I'll split that into its own PR.

File: Makefile (hunk 1 above)
Comment (sftim) [reply Makefile:47]: Good idea

File: layouts/_default/baseof.html (hunk 3 above)
Comment (sftim) [reply layouts/_default/baseof.html:4]: (the comment should appear in the published HTML, but doesn't render into the page)

Comment (kbhawkey) [reply layouts/_default/baseof.html:4]: Ah, okay.

File: Makefile (hunk 1 above)
Comment (kbhawkey) [reply Makefile:47]: Perhaps the module-check is not needed.

File: netlify.toml [hunk 4]
```diff
@@ -20,17 +20,10 @@ HUGO_ENABLEGITINFO = "true"
 command = "git submodule update --init --recursive --depth 1 && make deploy-preview"
 
 [context.branch-deploy]
-command = "git submodule update --init --recursive --depth 1 && make deploy-preview"
+command = "git submodule update --init --recursive --depth 1 && make non-production-build"
```
Comment (kbhawkey) [inline netlify.toml:23]: When is the `branch-deploy` context used?

**Comment by kbhawkey** [comment]
👋 @sftim. The changes look okay. How can we test the changes?

**Comment by sftim** [comment]
I can't think of a good way to test this prior to making it actually live. If it's approved with a hold, I'm happy to shepherd this through and verify it and then revert if the changes are a problem. If you want to merge this and do the checks @kbhawkey that's fine too.

The key things to check are the `/robots.txt` for the live site (post-merge), the `<meta name="robots" content="...">` element in each page, and the HTTP headers for served pages.

Ideally this change also improves things by making other sites _not_ indexable, but if it doesn't we're no worse off.

**Comment by sftim** [comment]
@onlydole would you be willing to cast eyes over this change as well? It'd be unfortunate to make an error here!

**Comment by onlydole** [comment]
Howdy, @sftim, of course! 

I took a look at this and all these changes look straightforward to me. I do like how you set up the preview markup within the page and can't wait to see the index change performance! 

/lgtm

**Comment by sftim** [comment]
Thanks. I'll plan some time to roll this out and verify that it works as expected.

File: netlify.toml (hunk 4 above)
Comment (sftim) [reply netlify.toml:23]: The context was called `branch-deploy` before; I assume that's correct (I'm not touching that aspect).

**Comment by sftim** [comment]
/hold cancel

:crossed_fingers:

**Pull request merged**